src/Uno.UWP/Extensions/PermissionHelper.Android.cs

```diff
@@ -0,0 +1,198 @@
+﻿#if __ANDROID__
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.Extensions
+{
+	public partial class PermissionsHelper
+	{
+		private static TaskCompletionSource<bool> _permissionCompletionSource;
+
+		/// <summary>
+		/// Return null on error, or array of permission to be asked for (not granted at this time). Both parameters can be null
+		/// </summary>
+		/// <param name="requiredPermissions">Permissions that are required. Can be null.</param>
+		/// <param name="optionalPermissions">Permissions that are required only if declared in Manifest. Can be null.</param>
+		/// <returns>Array of all permission that is not granted at this moment (can be empty!), or null if some error occured</returns>
+		public static string[] MissingPermissions(string[] requiredPermissions, string[] optionalPermissions)
+		{
+			// since API 29, we should do something more:
+			// https://developer.android.com/reference/android/content/pm/PackageInstaller.SessionParams.html#setWhitelistedRestrictedPermissions(java.util.Set%3Cjava.lang.String%3E)
+
+			// do we have declared permissions in Manifest?
+			Android.Content.Context context = Android.App.Application.Context;
+			Android.Content.PM.PackageInfo packageInfo =
+				context.PackageManager.GetPackageInfo(context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			var manifestPermissions = packageInfo?.RequestedPermissions;
+			if (manifestPermissions is null)
+				return null;
+
+			// test required permissions

```

Comment:
This method should be split in smaller parts. If you're adding a comment that explains roughly what the code below does, it's generally a method name. Here, it's `TestRequiredPermissions`, same below. This will improve your code readability.